e2e/specs/translation-builder-workflows.spec.ts

```diff
@@ -0,0 +1,178 @@
+    formUuid = form.uuid;
+
+    const valueReference = await createValueReference(api);
+    await addFormResources(api, valueReference, formUuid);
+  });
+  test('Manage translations: switch languages, filter, and search', async ({ page }) => {
+    const formBuilderPage = new FormBuilderPage(page);
+
+    await test.step('When I open a form in the translation builder', async () => {
+      await formBuilderPage.gotoFormBuilder();
+      await formBuilderPage.searchForForm(formDetails.name);
+      await formBuilderPage.page.getByRole('row', { name: formDetails.name }).getByLabel('Edit Schema').first().click();
+      await formBuilderPage.translationBuilderTab().click();
+    });
+
+    await test.step('When I open the language dropdown', async () => {
+      await formBuilderPage.languageDropdown().click();
+    });
+
+    await test.step('Then I should see the language list', async () => {
+      const dropdownMenu = formBuilderPage.translationBuilderPanel().locator('.cds--list-box__menu');
+      await expect(dropdownMenu).toBeVisible();
+    });
+
+    await test.step('And I should see "English (en)" in the language list', async () => {
+      await expect(formBuilderPage.page.getByRole('option', { name: 'English (en)' })).toBeVisible();
+    });
+
+    await test.step('When I select the "All" translations tab', async () => {
+      await formBuilderPage.allTranslationsTab().click();
+    });
+
+    await test.step('Then the "All" translations tab should be selected', async () => {
+      await expect(formBuilderPage.allTranslationsTab()).toHaveAttribute('aria-selected', 'true');
+    });
+
+    await test.step('When I select the "Translated" translations tab', async () => {
+      await formBuilderPage.translatedTab().click();
+    });
+
+    await test.step('Then I should see 0 "translated" entries', async () => {
+      await expect(page.locator('[data-status="translated"]')).toHaveCount(0);
+    });
+
+    await test.step('When I select the "Untranslated" translations tab', async () => {
+      await formBuilderPage.untranslatedTab().click();
+    });
+
+    await test.step('Then the "Untranslated" translations tab should be selected', async () => {
+      await expect(formBuilderPage.untranslatedTab()).toHaveAttribute('aria-selected', 'true');
+    });
+
+    await test.step('And I should see at least 1 "untranslated" entry', async () => {
+      await expect(page.locator('[data-status="untranslated"]').first()).toBeVisible();
+    });
+
+    await test.step('When I search translations for "Visit Details"', async () => {
+      const searchInput = formBuilderPage.translationSearchInput();
+      await searchInput.fill('Visit Details');
+    });
+
+    await test.step('Then the translation search input should contain "Visit Details"', async () => {
+      const searchInput = formBuilderPage.translationSearchInput();
+      await expect(searchInput).toHaveValue('Visit Details');
```

Comment:
chore: Prefer to use regex to search for strings. Lets update other similar instances as well
```suggestion
      await expect(searchInput).toHaveValue(/visit details/i');
```